File: coins/src/main/java/com/bubelov/coins/database/DbHelper.java
package com.bubelov.coins.database;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import com.bubelov.coins.R;
import com.bubelov.coins.service.sync.merchants.MerchantsSyncService;

/**
 * Author: Igor Bubelov
 * Date: 07/07/14 22:18
 */

public class DbHelper extends SQLiteOpenHelper {
    private Context context;

    public DbHelper(Context context) {
        super(context,
                context.getResources().getString(R.string.database_name),
                null,
                context.getResources().getInteger(R.integer.database_version));
        this.context = context;
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        for (String statement : new AssetSqlQuery(context, "create_tables").getStatements()) {
            db.execSQL(statement);
        }
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        for (String statement : new AssetSqlQuery(context, "drop_tables").getStatements()) {
            db.execSQL(statement);
        }

        context.startService(MerchantsSyncService.makeIntent(context, true));
    }
}

File: app/src/main/java/com/bubelov/coins/db/DbHelper.kt
package com.bubelov.coins.db

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

/**
 * @author Igor Bubelov
 */

class DbHelper(context: Context) : SQLiteOpenHelper(context, DbHelper.DATABASE_NAME, null, DbHelper.DATABASE_VERSION) {
    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(CREATE_PLACES__SQL)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL(DROP_PLACES_SQL)
        onCreate(db)
    }

    companion object {
        private val DATABASE_NAME = "database.db"

        private val DATABASE_VERSION = 1

        private val CREATE_PLACES__SQL =
                "create table places (\n" +
                        "    _id integer primary key,\n" +
                        "    _updated_at integer not null,\n" +
                        "    latitude real not null,\n" +
                        "    longitude real not null,\n" +
                        "    name text not null,\n" +
                        "    description text not null,\n" +
                        "    phone text not null,\n" +
                        "    website text not null,\n" +
                        "    category_id integer not null,\n" +
                        "    opening_hours text not null,\n" +
                        "    visible boolean not null,\n" +
                        "    currencies text not null,\n" +
                        "    opened_claims integer not null,\n" +
                        "    closed_claims integer not null\n" +
                        ");"

        private val DROP_PLACES_SQL = "drop table if exists places;"
    }
}